x-pack/filebeat/input/http_endpoint/input.go

```diff
@@ -346,6 +346,7 @@ func newHandler(ctx context.Context, c config, prg *program, pub func(beat.Event
 			hmacKey:      c.HMACKey,
 			hmacType:     c.HMACType,
 			hmacPrefix:   c.HMACPrefix,
+			maxBodySize:  -1,
```

Comment:
@efd, just a from the perspective of design philosophy, since we are not allowing negative values in the config for **maxBodySize**, and any **<0** value is considered as an error, wouldn't it be more in line to keep the default as nil using a pointer instead of -1 ?

Reply:
This field is internal. The reason for using a negative instead of an optional pointer is that it's just a sign check rather than a pointer check and deref.